dev_setup.sh

```diff
@@ -0,0 +1,142 @@
+#!/bin/sh
+
+# cryptol setup script
+#
+# Supported distribution(s):
+#  * macOS 14 (AArch64)
```

Comment:
This mentions MacOS 14, but not 12, despite 12 being mentioned in comments elsewhere. Is that deliberate?

Reply:
I updated the docs throughout. The current state is
- Supported distribution is macOS 14
- I left in all the conditionals and checks for macOS 12
- I added a top-level check [per your next comment](https://github.com/GaloisInc/cryptol/pull/1674#discussion_r1622691534) to reject anyone running macOS 12

Hopefully the next dev that wanders in with macOS 12 can uncomment the top-level check, confirm that the rest of the script works, and not have to do any additional work.